### ISSUE TITLE
[1.17.x] [UX] Enlarge Sidebar UI text size and/or elements

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#6663]
### Describe the desired feature

The sidebar UI text size is now a bit too small to read (and leaves a lot of empty space) on fullscreen mode as compared to the 1.16 client. Both are using the defaulted display settings for windows 10. See attached screenshots. 

Wesnoth 1.17.3+dev
![image_a](https://user-images.githubusercontent.com/5283677/166123088-0c955d8b-df19-4ed7-acf1-72bfd9bd9001.png)

Wesnoth 1.16.2
![image_b](https://user-images.githubusercontent.com/5283677/166123090-9d367d68-6909-4494-adb1-0d9039018ce8.png)

**Comment by Wedge009** [comment]
The images you've posted suggests there's some scaling increase done in 1.16.2 and 1.17.3+dev is rendered at 100%. I have the inverse problem/behaviour: I just tested 1.17.3+dev (f17b8b76e98) and it looks horribly over-scaled up, while 1.16.2+dev and 1.16.2 release both look nicely rendered at 100%. What's going on? Did something change in master in the last 24 hours?

**Comment by Zireael07** [comment]
#6629 strikes me as a possible culprit (GH says 'merged yesterday')

**Comment by Wedge009** [comment]
Yes, I remembered seeing that, didn't realise it was merged already. @mesilliac Any idea if @knyghtmare's observations might be related to your changes since it looks like my observations are intended.

For me, the result is horrible. I know everyone with high-DPI/resolution screens moans about how tiny everything looks but at 2560x1440 I feel like I'm playing on something as squished as 800x600...

**Comment by mesilliac** [comment]
@Wedge009 yeah by default it tries to match the default window resolution for Wesnoth, which is set in code to be 1280x720. You can disable it in preferences by setting the pixel scale to 1.

We're still working on the high-dpi support, so this is an initial step towards it. The next step will likely bring smooth fonts and images at all scales.

As far as this issue goes (and also going by some of the discussion on IRC), there are two problems here:
 1. the default Wesnoth theme doesn't handle resolutions of 1920x1080 and above well at all. Everything becomes way too tiny;
 2. fractional scaling is not enabled, so if you're using fractional scaling at the OS level it won't be possible to match the previous scale factor with this new pixel scaling code, which currently only allows integer scale factors.

(1) needs somebody who can work on themes to look at it. The smaller themes also need some work, and had some bugs even before the pixel scaling PR.

(2) will be easier to fix once high-dpi support progresses a bit. Personally i find the result of fractional scaling unacceptable as it blurs all the beautiful pixel art, but perhaps for others this is an acceptable tradeoff for having a well-sized display. However, ideally, if UI scale problems like this can be fixed with theme work, they should be.

**Comment by mesilliac** [comment]
There also might be the possibility of some sort of separate "UI Scale" option, like many games have. But i have no idea how doable this is. I haven't looked at how themes work at all, so i don't know if it makes sense to scale them.

**Comment by Wedge009** [comment]
I recognise I'm a minority, but I'm actually quite comfortable with 1x scaling at 2560x1440 and above. But I probably don't have the very high DPI screens that many of you are using.

That aside, does what @knyghtmare posted above make sense? I wonder if the images are swapped between master and 1.16.2, possibly.

**Comment by mesilliac** [comment]
It makes sense, he has 125% OS scaling, but with this patch the OS scaling is disabled because the executable is now marked as HiDPI aware. So the OS was previously stretching the whole window to 125% as a compatibility measure, but now displays at 100% because it trusts the app to handle the scaling itself.

And yeah i understand that the higher resolution will be preferable for many people. I set it to scale by default mostly for the initial impression of new players. I feel it's a better first impression to appear in lower resolution than with tiny UI. Hopefully the low-res feel won't be so bad once fonts are rendered more nicely, but it's still open to feedback and tweaking.

**Comment by knyghtmare** [comment]
This is how a medium sized map now appears for me. Wesnoth 1.17.3+dev client.
![image](https://user-images.githubusercontent.com/5283677/166149922-de9bc2e9-d738-4bd0-9026-100ebcac96c0.png)

I can adjust the gameboard by zooming in once...but the side bar...i cannot adjust it...
I wear glasses...and even wearing the pair, this sidebar is too small to read. Thus, I labelled it as a bad UX problem.

**Comment by knyghtmare** [comment]
Also, I found a solution for this:
![image](https://user-images.githubusercontent.com/5283677/166150273-2780793d-29bf-45dc-aee6-f4aa480ff8c2.png)
I just let Windows tell the application to not scale itself anymore

**Comment by Vultraz** [comment]
8e3105dae6bf8c0de07e8630e44cee1db3db967d and 43935380130d0f29ec45d89db415d1b03e458748 are supposed to override Windows' automatic scaling. It works here, but I'm not sure why it doesn't for everyone... I did need to make changes to `packages/windows/wesnoth.rc` when compiling to make the changes to the manifest file apply to the output exe, so maybe that has something to do with it...?

**Comment by mesilliac** [comment]
For now, overriding to use OS scaling is the correct workaround for this.

It would be possible to add fractional scaling, but the effect would be identical to OS scaling. As such it would be better to spend that development time on either improving the 1920x1080 UI theme, or implementing global UI scaling.

I'm not sure how viable global UI scaling would be. There's another issue tracking that specifically at #6547 

Easier would be to implement this as part of an upgrade of the sidebar to GUI2. This is slated to happen at some point in the future, but doesn't have a definite timeline.

The short-term solution would be to design a better 1920x1080 theme in the current system. This would be a good first issue for somebody who can code WML (or wants to learn). I made a new issue for this: https://github.com/wesnoth/wesnoth/issues/6774

**Comment by CelticMinstrel** [comment]
The old `font_scale` setting in preferences might also help with this issue, if you set it to 125 or so.

**Comment by Pentarctagon** [comment]
> I made a new issue for this: https://github.com/wesnoth/wesnoth/issues/6774

Can this issue be closed then?

**Comment by mesilliac** [comment]
I think as it now it counts as a duplicate of #6547 , although the comments there are mainly about text only

**Comment by cooljeanius** [comment]
> I think as it now it counts as a duplicate of #6547 , although the comments there are mainly about text only

Hm, I'd think it seems like more of a duplicate of #2454 personally, but I dunno...

> The old `font_scale` setting in preferences might also help with this issue, if you set it to 125 or so.

That's what you added back in 2de3246, right?

**Comment by cooljeanius** [comment]
version string in issue title should be bumped, IMO
